### PR TITLE
Remove DeviceCheck and SafetyNet

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-93

### CODE REVIEW

File: migrations/000028_drop_devicecheck_and_safetynet.up.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+BEGIN;
+
+ALTER TABLE AuthorizedApp
```
Comment (sethvargo) [inline migrations/000028_drop_devicecheck_and_safetynet.up.sql:17]: 🛑 ⚠️ We're dropping all this data.

File: docs/index.md [hunk 2]
```diff
@@ -22,8 +22,8 @@ The Exposure Notification Server is responsible for the following functions:
 
 * Accepting the temporary exposure keys of affected users from mobile devices.
 
-* Validating the temporary exposure keys using the
-[device attestation API](https://developer.android.com/training/safetynet/attestation).
+* Validating the temporary exposure keys using a configurable third-party
+  verification service.
```
Comment (mikehelmick) [inline docs/index.md:26]: Link to the design for the certificate validation?

File: docs/server_functional_requirements.md [hunk 3]
```diff
@@ -81,19 +81,14 @@ Minimum required fields, followed by a JSON example:
   * Description: 2 letter country to identify the region(s) a key is valid for.
 * `appPackageName` (**REQUIRED**)
   * Type: string
-  * Constraints:
-    * For Android device, this must match the device `appPackageName` in the
-      SafetyNet verification payload
   * Description: name of the application bundle that sent the request.
 * `platform` (**REQUIRED**)
```
Comment (mikehelmick) [inline docs/server_functional_requirements.md:85]: Platform is also deprecated now - it was only needed to differentiate between DC and SN calls.

File: migrations/000028_drop_devicecheck_and_safetynet.up.sql (hunk 1 above)
Comment (mikehelmick) [reply migrations/000028_drop_devicecheck_and_safetynet.up.sql:17]: maybe put that in a comment in the migration?

File: tools/admin-console/templates/authorizedapp.html [hunk 4]
```diff
@@ -35,89 +35,6 @@ <h2>Edit Application {{.app.AppPackageName}}</h2>
   </div>
   <hr/>
 
-  <div class="alert alert-info" role="alert">For Android Applications, these settings define if SafetyNet device attestation is in use and how it should be enforced.</div>
```
Comment (mikehelmick) [inline tools/admin-console/templates/authorizedapp.html:38]: all my HTML 😢

File: migrations/000028_drop_devicecheck_and_safetynet.up.sql (hunk 1 above)
Comment (sethvargo) [reply migrations/000028_drop_devicecheck_and_safetynet.up.sql:17]: Done

File: docs/index.md (hunk 2 above)
Comment (sethvargo) [reply docs/index.md]: Done

File: tools/admin-console/templates/authorizedapp.html (hunk 4 above)
Comment (sethvargo) [reply tools/admin-console/templates/authorizedapp.html:38]: 👋

File: docs/server_functional_requirements.md (hunk 3 above)
Comment (sethvargo) [reply docs/server_functional_requirements.md]: Done

**Comment by sethvargo** [comment]
@mikehelmick just removed platform. Please re-review.

**Comment by mikehelmick** [comment]
> @mikehelmick just removed platform. Please re-review.

thanks!

/lgtm
/approve

**Comment by jexh** [comment]
Hi, I am just wondering why you decided to drop these to resolve GH-93, instead of fixing the issue? I was planning some changes in this area that I wanted to suggest. Surely SafetyNet and DeviceCheck still add valuable protections here?

**Comment by sethvargo** [comment]
We're moving verification to third-party PHA servers. PHAs are free to use DC/SN to verify devices, but the server will soon move to using the `verificationPayload`. See https://github.com/google/exposure-notifications-server/blob/master/docs/design/verification_protocol.md and #508

**Comment by jexh** [comment]
OK, thanks. That makes a lot of sense. As a follow up question: is there a plan to open source a reference PHA implementation?

**Comment by sethvargo** [comment]
I can't discuss future plans, but that's definitely a possibility.

**Comment by jexh** [comment]
Thanks again, my last comment on this point. For my day job, I help to develop Approov(approov.io) a service for verifying communications are coming from authentic untampered apps (includes DC and SN options as well as other checks) with a simple flow. My changes were going to offer a generic JWT verification alternative alongside the DC/SN flow to support 3rd party app-verification flows like ours. If you do release a reference PHA implementation in the future, would you be open to including changes to this area?

**Comment by sethvargo** [comment]
@mikehelmick is owning the PHA verification, so I'll let him comment.

**Comment by mikehelmick** [comment]
The plan is to open source a reference PHA implementation.

At this time, I can't comment on what technologies we may use / not use for that service. It will likely be up to individual operators (governments / public health authorities) when the time comes.

Another motivation for removing device check / safetynet is to avoid communication with Google and Apple servers for purposes of exposure notifications. It eliminates the perception that Google and or Apple could be identifying devices that are reporting infections.

**Pull request merged**